### PR TITLE
Address false positives seen with argocd, grafana, jupyterhub, and reflex

### PR DESCRIPTION
**Pull request by egibs**
This PR addresses the [critical] false positives seen in these PRs:
- https://github.com/wolfi-dev/os/pull/29762
- https://github.com/wolfi-dev/os/pull/29784

The difficult exclusion was with the `PrintDeps.exe` file, so I opted to exclude this string:
```
Usage:
  PrintDeps FILE...     Version: r  %p  eE  pP
```

Depends on https://github.com/chainguard-dev/malcontent-samples/pull/8

### CODE REVIEW

**Comment by egibs** [comment]
The changes in [`bd8a2d8` (#475)](https://github.com/chainguard-dev/malcontent/pull/475/commits/bd8a2d8e3f4214ec1d38d714afb86f6aae807c32) are based on eight different `.js` files which would be tedious to maintain as samples. The third-party rule was the `argocd` finding.

File: rules/obfuscation/js/char_codes.yara [hunk 1]
```diff
@@ -13,9 +13,12 @@ rule child_process : critical {
 		$a_shift = "shift"
 		$a_push = "push"
 
-		$const = "const "
-		$function = "function("
-		$return = "{return"
+		$a_const = "const "
+		$a_function = "function("
+		$a_return = "{return"
+
```
Comment (tstromberg) [inline rules/obfuscation/js/char_codes.yara:19]: also mark this as a "high"

File: rules/combo/degrader/edr_killer.yara [hunk 2]
```diff
@@ -28,8 +30,10 @@ rule win_kill_proc : critical {
 	$debug_uhf = "UnhandledExceptionFilter"
 	$kill_gmh = "GetModuleHandle"
 	$kill_tp = "TerminateProcess"
+
+    $not_printdeps_exe = {55 73 61 67 65 3A 0A 20 20 50 72 69 6E 74 44 65 70 73 20 46 49 4C 45 2E 2E 2E 00 00 00 00 00 56 65 72 73 69 6F 6E 3A 20 72 00 00 25 70 00 00 65 45 00 00 70 50}
```
Comment (tstromberg) [inline rules/combo/degrader/edr_killer.yara:34]: Also change criticality to high

File: rules/combo/stealer/browser.yara [hunk 3]
```diff
@@ -60,6 +60,8 @@ rule multiple_browser_refs : critical {
     $not_ff_js = "Firefox can even throw an error"
     $not_generated_comment = "// This file is generated"
     $not_generated_file = "/utils/generate_types/index.js"
+    $not_microsoft = "Copyright (c) Microsoft Corporation."
```
Comment (tstromberg) [inline rules/combo/stealer/browser.yara:63]: Also change criticality to high

File: rules/combo/stealer/crypto.yara [hunk 4]
```diff
@@ -11,6 +11,8 @@ rule js_crypto_stealer : critical {
     $url = /https{0,1}:\/\/[\w][\w\.\/\-_\?=\@]{8,64}/
 
 	$POST = "POST"
+
+	$not_grafana = "self.webpackChunkgrafana=self.webpackChunkgrafana"
   condition:
-	filesize < 50KB and $url and $POST and any of ($pk*)
+	filesize < 50KB and $url and $POST and any of ($pk*) and none of ($not_*)
```
Comment (tstromberg) [inline rules/combo/stealer/crypto.yara:17]: Also change criticality to high

File: rules/obfuscation/js/function_spam.yara [hunk 5]
```diff
@@ -5,6 +5,21 @@ rule js_const_func_obfuscation : critical {
...
+		$not_grafana2 = "@grafana/data"
+		$not_grafana3 = "@grafana/runtime"
+		$not_grafana4 = "@grafana/ui"
+		$not_lodash = "Released under MIT license <https://lodash.com/license>"
+		$not_lodash2 = {4C 6F 64 61 73 68 20 3C 68 74 74 70 73 3A 2F 2F 6C 6F 64 61 73 68 2E 63 6F 6D 2F 3E}
+		$not_mit = "This source code is licensed under the MIT license found in the * LICENSE file in the root directory of this source tree."
+		$not_openjs = "Copyright OpenJS Foundation and other contributors <https://openjsf.org/>"
 	condition:
-		filesize < 256KB and #const > 32 and #function > 48 and #return > 64
+		filesize < 256KB and #const > 32 and #function > 48 and #return > 64 and none of ($not_*)
```
Comment (tstromberg) [inline rules/obfuscation/js/function_spam.yara:24]: Change criticality to medium

File: rules/obfuscation/js/parseInt.yara [hunk 6]
```diff
@@ -7,6 +7,8 @@ rule js_const_func_obfuscation : critical {
 		$function = "function("
 		$return = "{return"
 		$parseInt = "parseInt"
+
+		$not_grafana = "self.webpackChunkgrafana=self.webpackChunkgrafana"
 	condition:
-		filesize < 256KB and #const > 16 and #function > 32 and #parseInt > 8 and #return > 32
+		filesize < 256KB and #const > 16 and #function > 32 and #parseInt > 8 and #return > 32 and none of ($not_*)
```
Comment (tstromberg) [inline rules/obfuscation/js/parseInt.yara:13]: Change criticality to high

**Review by tstromberg — CHANGES_REQUESTED** [review]

File: rules/obfuscation/js/char_codes.yara (hunk 1 above)
Comment (egibs) [reply rules/obfuscation/js/char_codes.yara:19]: If we move these to high, I can remove the exclusions. That would be a lot cleaner.

Comment (tstromberg) [reply rules/obfuscation/js/char_codes.yara:19]: Works for me. In my mind, "CRITICAL" rules should never be broad enough for false positives to happen.

**Review by tstromberg — APPROVED** [review]

**Pull request merged**